usrp_drivers/n200/n200_driver.cpp

```diff
@@ -382,245 +398,253 @@ std::string random_string( size_t length )
-      }
-      if (md.get_error_code() == uhd::rx_metadata_t::ERROR_CODE_OVERFLOW) {
-          // TODO(keith): throw error
-      }
-      if (md.get_error_code() != uhd::rx_metadata_t::ERROR_CODE_NONE) {
           // TODO(keith): throw error
-      }
+        case uhd::rx_metadata_t::ERROR_CODE_OVERFLOW :
+          break;
+          //TODO(keith): throw error
+        default :
+          break;
+          //TODO(keith): throw error
+      } // REPLY I dont think there is a performance issue here, but i reworked this if you think it could look cleaner. We still need to compare against specific errors.
 
+      //TODO(Keith): investigate status to see under what circumstances this could fail. Will need
+      //additional info sent to radar control if true.
+
+      // REVIEW #6 TODO: Handle fragmentation flag being set in metadata. If this happens that means the buffer was too small to hold the number of samples (shouldn't happen, so this would be an error)
+      // REPLY We can talk about this. I don't think this is an error
       accumulated_received_samples += num_rx_samps;
-      std::cout << "Accumulated received samples " << accumulated_received_samples <<std::endl;
+      DEBUG_MSG("Accumulated received samples " << accumulated_received_samples);
     }
 
-    std::cout << "RECEIVE received samples per channel " << accumulated_received_samples << std::endl;
+    DEBUG_MSG("RECEIVE received samples per channel " << accumulated_received_samples);
 
-    std::chrono::steady_clock::time_point recv_end = std::chrono::steady_clock::now();
-    std::cout << "RECEIVE receive timing: "
-      << std::chrono::duration_cast<std::chrono::microseconds>
-                                                  (recv_end - recv_begin).count()
-      << "us" << std::endl;
+    auto recv_end = std::chrono::steady_clock::now();
 
+    time_diff = std::chrono::duration_cast<std::chrono::microseconds>
+                                                  (recv_end - recv_begin).count();
+    DEBUG_MSG("RECEIVE receive timing: " << time_diff << "us");
 
-    std::chrono::steady_clock::time_point send_begin = std::chrono::steady_clock::now();
 
-    cp.set_numberofreceivesamples(dp.numberofreceivesamples());
-    cp.set_name(shr_mem_name);
-    cp.set_sequence_num(dp.sequence_num());
-    std::string cp_str;
-    cp.SerializeToString(&cp_str);
+    auto send_begin = std::chrono::steady_clock::now();
 
-    zmq::message_t cp_size_message(cp_str.size());
-    memcpy ((void *) cp_size_message.data (), cp_str.c_str(), cp_str.size());
+    rxsamplesmetadata::RxSamplesMetadata samples_metadata;
+    samples_metadata.set_numberofreceivesamples(driver_packet.numberofreceivesamples()); // REVIEW #0 Should this be accumulated_received_samples ?
+    samples_metadata.set_shrmemname(shr_mem_name);                                       // REPLY The recv loop will not break until recving until that # of samples has been received. We want this exact number of samples to process.
```

Comment:
It's possible that we would have more than the number of samples we want since the while statement has a < comparison.